### PR TITLE
Feature/partlyfixed

### PR DESCRIPTION
**Pull request by mlooz**
A draft for a meta problem that allows to set some arguments to fixed values and thus reduce the dimension of the search space. Still in progress, not marked as such to trigger the CI

### CODE REVIEW

**Comment by bluescarni** [comment]
@mlooz thanks for the PR, I think this is a great addition.

I was wondering what you think about a simplification in the constructor API. Instead of writing, e.g.,

```python
fixed_arguments=[1], fixed_flags=[True, False, False]
```

would it be feasible to have something like

```python
fixed_arguments=[1, None, None]
```

instead?

I have a few more minor stylistic points when you think this is ready, but I think it is already looking good.

**Comment by mlooz** [comment]
Good point, @bluescarni ! 
I switched to the simpler format, I was already using it in some scripts.

Thanks for the feedback, what stylistic points do you have in mind?

**Comment by mlooz** [comment]
The only failing test now is the doctest, which fails because https://projects.coin-or.org/ is unresponsive. I suggest replacing it with https://coin-or.github.io/Ipopt/

File: pygmo/_py_problems.py [hunk 1]
```diff
@@ -265,3 +265,158 @@ def get_decorator(self, fname):
...
+            # the input object).
+            self._problem = deepcopy(prob)
+        else:
+            # Otherwise, we attempt to create a problem from it. This will
+            # work if prob is an exposed C++ problem or a Python UDP.
+            self._problem = problem(prob)
+
+        minBound, maxBound = self._problem.get_bounds()
+
+        inner_dim = len(minBound)
```
Comment (bluescarni) [inline pygmo/_py_problems.py:330]: I am 99% sure that you cannot construct a ``pygmo.problem`` if inconsistent sizes are detected in the UDP's lower/upper bounds, so I believe this check is unnecessary.

Also, you can get the problem dimension directly from the ``pygmo.problem`` object, without querying for the size of the bounds.

File: pygmo/_py_problems.py [hunk 2]
```diff
@@ -265,3 +265,158 @@ def get_decorator(self, fname):
...
+        self.full_dim = inner_dim
+        if len(maxBound) != inner_dim:
+            raise ValueError("Problem bounds inconsistent!")
+
+        if len(fixed_arguments) != inner_dim:
+            raise ValueError("Got " + str(len(fixed_arguments)) + " argument array for problem of dimension " + str(inner_dim))
```
Comment (bluescarni) [inline pygmo/_py_problems.py:336]: Here and elsewhere I'd use the string formatting syntax:

```python
raise ValueError("Got {} argument array for problem of dimension {}".format(len(fixed_arguments), inner_dim))
```

File: pygmo/_py_problems.py [hunk 3]
```diff
@@ -265,3 +265,158 @@ def get_decorator(self, fname):
...
+        return self._problem.fitness(self.get_full_x(x))
+
+    def has_batch_fitness(self):
+        return self._problem.has_batch_fitness()
+
+    def batch_fitness(self, dvs):
+        contiguous_x = []
+        if len(dvs) % self.get_nx() != 0:
+            raise ValueError("Expected multiple of " + str(self.get_nx()) + " but got " + str(len(dvs)))
+        num_dvs = int(len(dvs) / self.get_nx())
```
Comment (bluescarni) [inline pygmo/_py_problems.py:395]: Can't you use the integer division operator here ``//``, rather than going through a float -> int conversion?

**Review by bluescarni — CHANGES_REQUESTED** [review]

**Comment by bluescarni** [comment]
@mlooz I have left a couple of comments in the review.

**Comment by mlooz** [comment]
@bluescarni Thanks, I addressed the comments

**Pull request merged**